doc/users/next_whats_new/fill_between_poly_collection.rst

```diff
@@ -0,0 +1,22 @@
+``FillBetweenPolyCollection``
+-----------------------------
+
+The new class :class:`matplotlib.collections.FillBetweenPolyCollection` provides
+the ``set_data`` method, enabling e.g. resampling
+(:file:`galleries/event_handling/resample.html`).
+:func:`matplotlib.axes.Axes.fill_between` and
+:func:`matplotlib.axes.Axes.fill_betweenx` now return this new class.
+
+.. code-block:: python
+
+    import numpy as np
+    from matplotlib import pyplot as plt
+
+    t = np.linspace(0, 1)
+
+    fig, ax = plt.subplots()
+    coll = ax.fill_between(t, -t, t)
```

Comment:
Let's do
```suggestion
    coll = ax.fill_between(t, -t**2, t**2)
```
which may be slightly easier to understand than a lot of ts (t, -t, t) and is structurally closer to the update `(t, -t**4, t**4)`.